teamup-client/src/app/core/services/auth/auth.service.ts

```diff
@@ -0,0 +1,43 @@
+import {Injectable} from '@angular/core';
+import {UserLogin} from '../../models/userLogin';
+import {HttpClient, HttpErrorResponse} from '@angular/common/http';
+import {Router} from '@angular/router';
+import {delay} from 'q';
+
+
+
+/**
+ * This is the authentication service. It handles everything related to permission and token management
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+  constructor(private http: HttpClient, private router: Router) {
+  }
+
+  /**
+   * This function logs in users and handles eventual errors.
+   * @param user that is going to be logged in
+   */
+  public login(user: UserLogin) {
+    console.log(user);
```

Comment:
I don`t think that you should subscribe in here, just store the token and return an observable. Also remove the console logs